### PR TITLE
Disable AI analysis for 2D and ND datasets

### LINKED ISSUE #7953: Disable AI models for 2D and ND datasets
## Context
Our AI models are configured and tuned for 3D EM datasets. It does not work well on 2D datasets (and was never intended for those).

Example segmentation of some 2D dataset:  Screenshot 2024-07-26 at 11 16 47

See https://scm.slack.com/archives/CMBMU5684/p1721985399703099?thread_ts=1721318145.073349&cid=CMBMU5684

## Expected Behavior
Probably add some validation before launching any of the AI features. 2D and ND datasets should probably be denied from running any AI inference. 

Perhaps also include guards against weird datatypes, e.g. `uint24`. IIRC, the AI inference only works with `uint8` (please double-check).

Have a look at the "Create Animation job" modal. That includes validation against 2D and ND datasets.

## Current Behavior
The AI job runs on the 2D dataset but the output is not useful. All voxels were merged into one segment per chunk and agglomerated into one single segment.

## Steps to Reproduce the bug

- [ ] Cannot reproduce the bug anymore / needs deeper investigation.

See https://scm.slack.com/archives/CMBMU5684/p1721985399703099?thread_ts=1721318145.073349&cid=CMBMU5684

## Your Environment for bug

- Version of WEBKNOSSOS (Release or Commit): Version: 28878 (Server is currently at 28881!)
- [ ] Specific to long-running jobs (set `jobsEnabled=true` in `application.conf`)
- [ ] Specific to webknossos.org (set `isDemoInstance=true` in `application.conf`)

### PR DESCRIPTION
**Pull request by dieknolle3333**
### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- test that "AI Analysis" button is disabled if dataset is in 2D, ND or only has color layers in uInt24 data format
- make sure that you cannot select color layers with dataformat uint24 in layer selection (neither train nor run)
- everything else should work as before :)

### TODOs:
- [x] disable for 2D and ND
- [x] error for weird units
- [x] maybe disable/filter out uint24 layer for train model?
- [x] find out whether uint24 blacklisted or uint8 whitelisted
- [x] test for 2D
- [x] test review changes, e.g. with DS without color layer

### Issues:
- fixes #7953

------
(Please delete unneeded items, merge only when none are left open)
- [x] Updated [changelog](../blob/master/CHANGELOG.unreleased.md#unreleased)
- [ ] Removed dev-only changes like prints and application.conf edits :exclamation: :exclamation: :boom: :boom:

### CODE REVIEW

File: conf/application.conf [hunk 1]
```diff
@@ -146,12 +146,12 @@ features {
   discussionBoard = "https://forum.image.sc/tag/webknossos"
   discussionBoardRequiresAdmin = false
   hideNavbarLogin = false
-  isWkorgInstance = false
```
Comment (dieknolle3333) [inline conf/application.conf:149]: och nöööööööö

Comment (dieknolle3333) [reply conf/application.conf:149]: remove me

File: frontend/javascripts/oxalis/view/action_bar_view.tsx [hunk 2]
```diff
@@ -272,7 +270,34 @@ class ActionBarView extends React.PureComponent<Props, State> {
...
+    const datasetHasOnlyUInt24ColorLayers = colorLayersRightDataFormat.length === 0;
+    const isNd = (colorLayers[0].additionalAxes ?? []).length > 0;
+    const is2DOrNDDataset = isNd || is2d;
+    const isAIAnalysisDisabled = !getIsAIAnalysisEnabled();
+    const shouldDisableAIJobButton =
+      isAIAnalysisDisabled ||
+      datasetHasNoColorLayer ||
+      is2DOrNDDataset ||
+      datasetHasOnlyUInt24ColorLayers;
+    const getAIIsDisabledTooltip = () => {
```
Comment (hotzenklotz) [inline frontend/javascripts/oxalis/view/action_bar_view.tsx:287]: Gibt's ein speziellen Grund, warum das ein Funktion ist? Kann doch auch einfach eine Variable sein? `let tooltip = "" ...`

File: frontend/javascripts/oxalis/view/action_bar_view.tsx [hunk 3]
```diff
@@ -272,7 +270,34 @@ class ActionBarView extends React.PureComponent<Props, State> {
       onDeleteLayout: this.handleLayoutDeleted,
     });
 
-    const datasetHasColorLayer = getColorLayers(dataset).length > 0;
+    const colorLayers = getColorLayers(dataset);
+    const colorLayersRightDataFormat = colorLayers.filter(
```
[the remark this answers is not in the document]
Comment (dieknolle3333) [reply frontend/javascripts/oxalis/view/action_bar_view.tsx:274]: das sorgt dafür dass der "AI Analysis" button disabled wird wenns nur solche color layer gibt. kanns auch einfach rausnehmen!

File: frontend/javascripts/oxalis/view/action_bar_view.tsx [hunk 4]
```diff
@@ -272,7 +270,29 @@ class ActionBarView extends React.PureComponent<Props, State> {
...
+      (layer) => layer.elementClass !== "uint24",
+    );
+    const datasetHasNoColorLayer = colorLayers.length === 0;
+    const datasetHasOnlyUInt24ColorLayers = colorLayersRightDataFormat.length === 0;
+    const isNd = (colorLayers[0].additionalAxes ?? []).length > 0;
```
Comment (hotzenklotz) [inline frontend/javascripts/oxalis/view/action_bar_view.tsx:279]: Wenn es keinen Color layer gibt, dann crashed WK hier. 

Screenshot 2024-07-31 at 16 00 50

**Review by hotzenklotz — APPROVED** [review]
🚀

**Pull request merged**